music_assistant/providers/phishin/helpers.py

```diff
@@ -0,0 +1,463 @@
+"""Helper functions for Phish.in provider."""
+
+from __future__ import annotations
+
+import contextlib
+from collections.abc import Callable
+from typing import TYPE_CHECKING, Any
+
+import aiohttp
+from music_assistant_models.enums import AlbumType, ContentType, ExternalID, ImageType, MediaType
+from music_assistant_models.errors import MediaNotFoundError, ProviderUnavailableError
+from music_assistant_models.media_items import (
+    Album,
+    Artist,
+    AudioFormat,
+    ItemMapping,
+    MediaItemImage,
+    MediaItemMetadata,
+    Playlist,
+    ProviderMapping,
+    Track,
+)
+from music_assistant_models.unique_list import UniqueList
+
+from music_assistant.controllers.cache import use_cache
+
+from .constants import (
+    API_BASE_URL,
+    FALLBACK_ALBUM_IMAGE,
+    PHISH_ARTIST_ID,
+    PHISH_ARTIST_NAME,
+    PHISH_DISCOGS_ID,
+    PHISH_MUSICBRAINZ_ID,
+    PHISH_TADB_ID,
+    REQUEST_TIMEOUT,
+)
+
+if TYPE_CHECKING:
+    from music_assistant.models.music_provider import MusicProvider
+
+
+@use_cache(expiration=3600)  # 1 hour
```

Comment:
so, instead of defining the cache for all requests, maybe consider setting it on each method seperate. That way you can play with the expiration. For example the details of one artist or album can have an expiration of something like 60 days, they wont likely change, while the expiration of a playlist-tracks listingneeds to be short lived as it can change dynamically

Reply:
This decorator removed. All methods now have individual cache durations with a justification comment